src/commands/admin/showcommands.ts

```diff
@@ -1,24 +1,40 @@
-import { Formatters, Message } from 'discord.js';
-import { botMasterPerms } from '@lib/permissions';
+import { ApplicationCommandOptionData, ApplicationCommandPermissionData, CommandInteraction, Formatters, Message } from 'discord.js';
+import { BOTMASTER_PERMS } from '@lib/permissions';
 import { Command } from '@lib/types/Command';
 
 export default class extends Command {
 
-	aliases = ['ls', 'listcmd'];
 	description = 'Show all commands, including disable commands.';
+	tempPermissions: ApplicationCommandPermissionData[] = [BOTMASTER_PERMS];
 
-	async permissions(msg: Message): Promise<boolean> {
-		return await botMasterPerms(msg);
-	}
+	options: ApplicationCommandOptionData[] = [{
+		name: 'restricted',
+		description: 'Use this argument to see the list of restricted commands.',
+		type: 'BOOLEAN',
+		required: false
+	}]
 
-	run(msg: Message): Promise<Message> {
+	async tempRun(interaction: CommandInteraction): Promise<void> {
+		const restricted = interaction.options.getBoolean('restricted');
+		if (restricted) {
+			/*
```

Comment:
Unneeded comment block?

Reply:
That's the code from my restricted commands branch, I left it there in case we decided we still wanted to do something with it but I'll get rid of it on here.